### PR TITLE
allowing user to specify cores

### PR DESCRIPTION
**Pull request by Shimwell**
## Proposed changes

This small change to the dockerfile allows the use of another --build-arg in the dockerfile

This build arg controls the number of cores the image is built with. 

Compiling with mutliple cores can speed up the MOAB compile step. 

Using -j is an option but sometimes this can result in other applications closing while the compile takes all of the cpu cores. 

Therefore this setting allows the user or config.yml to specify the number of cores used

Also the documentation at the top of the dockerfile has been updated

## Types of changes

What types of changes does your code introduce to the Paramak?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] Pep8 applied
- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,40 +1,50 @@
...
+# Run with the following command for a terminal notebook interface
 # docker run -it ukaea/paramak .
-
+#
 # Run with the following command for a jupyter notebook interface
 # docker run -p 8888:8888 ukaea/paramak /bin/bash -c "jupyter notebook --notebook-dir=/examples --ip='*' --port=8888 --no-browser --allow-root"
 
 
-# test with the folowing command
+# Once build the dockerimage cna be tested with the folowing command
```
Comment (RemDelaporteMathurin) [inline Dockerfile:39]: typos here 😃

Comment (Shimwell) [reply Dockerfile:39]:
```suggestion
# Once build the docker image can be tested with the following command
```

File: Dockerfile [hunk 2]
```diff
@@ -36,7 +36,7 @@
 # docker run -p 8888:8888 ukaea/paramak /bin/bash -c "jupyter notebook --notebook-dir=/examples --ip='*' --port=8888 --no-browser --allow-root"
 
 
-# Once build the dockerimage cna be tested with the folowing command
+# Once build the docker image can be tested with the following command
```
Comment (RemDelaporteMathurin) [inline Dockerfile:39]:
```suggestion
# Once built, the docker image can be tested with the following command
```

**Comment by Shimwell** [comment]
This has reduced the build docker and test duration by 2 mins

18 mins now https://github.com/ukaea/paramak/runs/1438224398?check_suite_focus=true
22 mins previously https://github.com/ukaea/paramak/runs/1438190323?check_suite_focus=true

**Pull request merged**